src/App.tsx

```diff
@@ -1,87 +1,96 @@
-import { useEffect, useState } from "react";
+import React, { useEffect, useState } from "react";
 import { remult } from "remult";
 import { Task } from "./shared/Task";
 import { TasksController } from "./shared/TasksController";
+import { observer } from "mobx-react-lite";
+import { action, makeAutoObservable, observable, runInAction } from 'mobx';
 
-const taskRepo = remult.repo(Task);
 
-async function fetchTasks(hideCompleted: boolean) {
-  return taskRepo.find({
-    limit: 20,
-    orderBy: { completed: "asc" },
-    where: { completed: hideCompleted ? false : undefined }
-  });
-}
 
 
+class Store {
+  taskRepo = remult.repo(Task);
+  readonly tasks = observable<Task>([]);
+  hideCompleted = false;
+  constructor() {
+    makeAutoObservable(this);
+  }
+  replaceTasks(t: Task[]) {
+    this.tasks.replace(t.map(t => makeAutoObservable({ ...t })));
+  }
+  async loadTasks() {
+    this.replaceTasks(await
+      this.taskRepo.find({
+        limit: 20,
+        orderBy: { completed: "asc" },
+        where: { completed: this.hideCompleted ? false : undefined }
+      }));
+  }
+  addTask() {
+    this.tasks.push(makeAutoObservable(new Task()))
```

Comment:
Doesn't it make more sense to move `makeAutoObservable` into `Task` so it'll be encapsulated there? It'll also remove the need for that in `replaceTasks`.

Reply:
👍